shared/sszutil/deep_equal.go

```diff
@@ -0,0 +1,193 @@
+package sszutil
```

Comment:
I realize there were no tests for this file specifically, but could you add a few sanity tests? Specifically, comparing protobuf generate structs with mixed byte slice values (nil vs empty slice are treated the same).

Reply:
Same, some tests would be great!

Reply:
I have added some test cases, especially for covering the `nil` vs `empty slice`. Feel free to comment.

While I am doing that for `struct`, I found that the `DeepEqual` will panic if the struct doesn't have any public fields, is that an issue? 

```
panic: reflect.Value.Interface: cannot return value obtained from unexported field or method [recovered]
	panic: reflect.Value.Interface: cannot return value obtained from unexported field or method
```